modules/gapi/test/s11n/gapi_s11n_tests.cpp

```diff
@@ -2,18 +2,52 @@
 
 #include "backends/common/serialization.hpp"
 
+namespace {
+    struct MyCustomType {
+        int val;
+        std::string name;
+        std::vector<float> vec;
+        std::map<int, uint64_t> mmap;
+        bool operator==(const MyCustomType& other) const {
+            return val == other.val && name == other.name &&
+                   vec == other.vec && mmap == other.mmap;
+        }
+    };
+}
+
+// FIXME: namespace conflicts
+namespace cv {
+namespace gapi {
+namespace s11n {
+namespace detail {
+    template<> struct S11N<MyCustomType> {
+        static void serialize(I::OStream &os, const MyCustomType &p) {
+            os << p.val << p.name << p.vec << p.mmap;
+        }
+        static MyCustomType deserialize(I::IStream &is) {
+            MyCustomType p;
+            is >> p.val >> p.name >> p.vec >> p.mmap;
+            return p;
+        }
+        static constexpr const bool isSupported = true;
```

Comment:
And this flag could be hidden with that macro too...

Reply:
Dropped for now. Can be done separately if will be needed